homeassistant/components/ambient_network/config_flow.py

```diff
@@ -0,0 +1,178 @@
+    ) -> FlowResult:
+        """Handle the initial step to select the location."""
+
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self._latitude = user_input[CONFIG_LOCATION][CONFIG_LOCATION_LATITUDE]
+            self._longitude = user_input[CONFIG_LOCATION][CONFIG_LOCATION_LONGITUDE]
+            self._radius = DistanceConverter.convert(
+                user_input[CONFIG_LOCATION][CONFIG_LOCATION_RADIUS],
+                UnitOfLength.METERS,
+                UnitOfLength.MILES,
+            )
+            return await self.async_step_station()
+
+        schema: vol.Schema = self.add_suggested_values_to_schema(
+            vol.Schema(
+                {
+                    vol.Required(
+                        CONFIG_LOCATION,
+                    ): LocationSelector(LocationSelectorConfig(radius=True)),
+                }
+            ),
+            {
+                CONFIG_LOCATION: {
+                    CONFIG_LOCATION_LATITUDE: self.hass.config.latitude,
+                    CONFIG_LOCATION_LONGITUDE: self.hass.config.longitude,
+                    CONFIG_LOCATION_RADIUS: CONFIG_LOCATION_RADIUS_DEFAULT,
+                }
+            },
+        )
+
+        return self.async_show_form(
+            step_id=CONFIG_USER,
+            data_schema=schema,
+            errors=errors,
+        )
+
+    @staticmethod
+    def get_station_name(station: dict[str, Any]) -> str:
+        """Pick a station name.
+
+        Station names can be empty, in which case we construct the name from
+        the location and device type.
+        """
+        if name := station.get(API_STATION_INFO, {}).get(API_STATION_NAME):
+            return str(name)
+        location = (
+            station.get(API_STATION_INFO, {})
+            .get(API_STATION_COORDS, {})
+            .get(API_STATION_LOCATION)
+        )
+        station_type = station.get(API_LAST_DATA, {}).get(API_STATION_TYPE)
+        return f"{location}{'' if location is None or station_type is None else ' '}{station_type}"
+
+    async def async_step_station(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the second step to select the station."""
+
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            mac_address, station_name = user_input[CONFIG_STATION].split(",")
+            await self.async_set_unique_id(mac_address)
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(
+                title=station_name,
+                data={
+                    ENTITY_STATION_NAME: station_name,
+                    ENTITY_MAC_ADDRESS: mac_address,
+                },
+            )
+
+        client: OpenAPI = OpenAPI()
+        stations: list[dict[str, Any]] = await client.get_devices_by_location(
+            self._latitude, self._longitude, radius=self._radius
+        )
+
+        # Filter out indoor stations
```

Comment:
Why would anyone ever add an indoor station to a weather station cloud service 😂

Reply:
Not sure. But it does happen.